### PR TITLE
improve GenProg analysis with ast-based search for statements in the source file

### PR DESCRIPTION
**Pull request by niloofartrg**

### CODE REVIEW

**Comment by niloofartrg** [comment]
Except for two failing test cases(the failing reason is like Arja failing test cases), the filter function works for all other files in the dataset. @monperrus

**Comment by monperrus** [comment]
remove npefix stuff

**Comment by niloofartrg** [comment]
extra files are deleted. @monperrus

File: src/test/java/fr/inria/coming/spoon/repairability/repairtools/NPEfixTest.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
     @Test
     public void NPEfixTest() throws Exception {
         FinalResult result = TestUtills.runRepairability("NPEfix", "/repairability_test_files/NPEfix/");
-        TestUtills.numberOfInstances(result, 18, 16);
```
Comment (niloofartrg) [inline src/test/java/fr/inria/coming/spoon/repairability/repairtools/NPEfixTest.java]: I have to change this back to the previous state(master state) if I have to delete the test files.

Comment (monperrus) [reply src/test/java/fr/inria/coming/spoon/repairability/repairtools/NPEfixTest.java]: What if you add again the NpeFix test files?

Comment (niloofartrg) [reply src/test/java/fr/inria/coming/spoon/repairability/repairtools/NPEfixTest.java]: added.

**Comment by monperrus** [comment]
Thank you!

**Pull request merged**